server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -493,6 +504,7 @@ public RecoverySettings(Settings settings, ClusterSettings clusterSettings) {
             CLUSTER_ROUTING_ALLOCATION_NODE_CONCURRENT_INCOMING_RECOVERIES_SETTING,
             this::setMaxConcurrentIncomingRecoveries
         );
+        clusterSettings.addSettingsUpdateConsumer(INDICES_RECOVERY_CHUNK_SIZE, this::setChunkSize);
```

Comment:
nit: the `setChunkSize` has a `// only settable for tests` comment that we can remove now

Reply:
Added the validation in https://github.com/elastic/elasticsearch/pull/114618/commits/c8625d88cf9962d53288e9168127755bc8fc9333